src/Npgsql/TypeHandlers/CompositeHandlers/CompositeMemberHandlerOfStruct.cs

```diff
@@ -81,9 +87,18 @@ public override async Task Write(TComposite composite, NpgsqlWriteBuffer buffer,
         }
 
         public override int ValidateAndGetLength(TComposite composite, ref NpgsqlLengthCache lengthCache)
-            => NullableHandler<TMember>.Exists
-                ? NullableHandler<TMember>.ValidateAndGetLength(_handler, _get(ref composite), ref lengthCache, null)
-                : _handler.ValidateAndGetLength(_get(ref composite), ref lengthCache, null);
+        {
+            if (_get == null)
+                ThrowHelper.ThrowNotSupportedException_NoPropertyGetter(typeof(TComposite), MemberInfo);
+
+            var value = _get(ref composite);
+            if (value == null)
```

Comment:
This looks like it corrects a bug we previously had when writing null members of composites? If so, should we add a test to exercise it?

Reply:
Done.